src/arrayMethodSort.js

```diff
@@ -4,8 +4,30 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function (compareFunction) {
     // write code here
+
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const x = this[i];
+
+            this[i] = this[j];
+            this[j] = x;
+          }
+        } else {
+          if (String(this[j]) < String(this[i])) {
```

Comment:
When converting elements to strings, it would be safer to use 'toString()' method explicitly. This will avoid issues with null or undefined values that do not have a 'String' constructor.